microprofile/cdi/src/main/java/io/helidon/microprofile/cdi/HelidonProxyServices.java

```diff
@@ -117,8 +136,20 @@ public boolean supportsClassDefining() {
                 // it also needs to open the package we are doing the lookup in
                 myModule.addReads(classModule);
             }
+
+            // I would like to create a private lookup in the same package as the proxied class, so let's
+            // try to load it
+            String proxiedClassName = className.substring(0, className.indexOf('$'));
```

Comment:
Thinking about it, this worries me.  If I have a nested class, won't this chop off the nested class name?  e.g. `com.foo.Bar$Baz$various$proxy$related$bits`: you'll conclude here that the proxied class name is `com.foo.Bar`, which is incorrect.  Or maybe this is intentional in some way, as the outer class (`Bar`) will of course be in the same package as the nested (and proxied) class (`Bar$Baz`) so it doesn't matter in practice?  Maybe a comment to further clarify here what you're doing?

Reply:
This is really to get a private lookup - so either you get the actual class, or the enclosing class - in both cases the private lookup should work

Reply:
Problem is that we only have a class that produces the bean, which is absolutely useless when in a different package.
I need to get a class that allows me to obtain the private lookup - this seemed to be the easiest approach, as it should work both for class itself or inner class.

Reply:
Then maybe the name is at fault?  Shouldn't it be `privateLookupClassName` or something?

Reply:
And if the class is not valid (e.g. we cannot load it), I fall back to the producer class name

Reply:
Also the method name is now `defineClassPrivateLookup` which is exactly what it does - it defines a class using a private lookup. 
The name is related to how the lookup is used, not how it is obtained.

Reply:
But isn't `proxiedClassName` the name of the lookup class, not the name of the class being proxied?  See line 148.

Reply:
Refactored the whole thing, closing this discussion. I hope the new approach is correctly named (and documented).